src/utils/include/utils/math/triangle_functions.hpp

```diff
@@ -77,10 +77,10 @@ double angle_btw_triangles(const T1 &P1, const T2 &P2, const T3 &P3,
     tmp11 = -std::sqrt(-tmp11);
   }
 
-  if (tmp11 >= 1.) {
-    tmp11 = 0.0;
-  } else if (tmp11 <= -1.) {
-    tmp11 = M_PI;
+  if (tmp11 > 1.) {
```

Comment:
Can you please explain these changes? Also I'm having a hard time understanding this function in general?
If you have any insight, can you please share it?

Reply:
Those few lines are only to guarantee  that tmp11 is from interval <-1,1>. I think that in some cases you may get tmp11 outside this interval.

In general, this method returns the angle between triangles p1,p2,p3 and p2,p3,p4. Tricky part is that the orientations of triangles must be taken into consideration. So the angle can be larger than 2 PI. This is done after the commented part by 

  tmp11 = -(normal1[0] * P1[0] + normal1[1] * P1[1] + normal1[2] * P1[2]);
  if (normal1[0] * P4[0] + normal1[1] * P4[1] + normal1[2] * P4[2] + tmp11 < 0)
    return 2 * M_PI - phi;

  return phi;

Reply:
Wouldn't it be more readable with e9a9068b32bb? Not sure why `oif_volume_conservation` fails randomly. Also, the corner case where `tmp11` is either -1 or +1 should be tested, because `acos(3.14)` was undefined.

Reply:
@jngrad I think what you are proposing is more readable, but there should be tests, and some docstring that shows what the function is doing, especially in the corner cases. I think for this utility function we should take specially care for quality.